packages/eventual-send/src/index.d.ts

```diff
@@ -18,6 +18,20 @@ export type EOnly<T> = T extends (...args: infer P) => infer R
     >
   : ERef<T>;
 
+type FilteredKeys<T, U> = {
+  [P in keyof T]: T[P] extends U ? P : never;
+}[keyof T];
+
+type DataOnly<T> = Omit<T, FilteredKeys<T, Function>>;
```

Comment:
Could you describe in english what `DataOnly` or `FunctionOnly` mean? Since I didn't understand `FilteredKeys` I can't figure out these. Also btw, what's `Omit` and `Pick`?

Reply:
`DataOnly<T>` means to return a record type `T2` consisting only of properties that are *not* functions.
`FunctionOnly<T>` means to return a record type `T2` consisting only of properties that are functions.

`Omit<T, U>` means to return a record type `T2` which has none of the properties whose keys are part of `U`.
`Pick<T, U>` means to return a record type `T2` which has only the properties whose keys are part of `U`.

`Omit<{a: 1, b: 2, c: 3}, 'b'>` is the type `{a: 1, c: 3}`.
`Pick<{a: 1, b: 2, c: 3}, 'b'>` is the type `{b: 2}`.